plugins/org.jkiss.dbeaver.ext.bigquery/src/org/jkiss/dbeaver/ext/bigquery/model/data/BigQueryTrace.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.bigquery.model.data;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.ext.bigquery.model.BigQueryResultSet;
+import org.jkiss.dbeaver.model.exec.trace.DBCTrace;
+import org.jkiss.utils.BeanUtils;
+import org.jkiss.utils.ByteNumberFormat;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class BigQueryTrace implements DBCTrace {
+    private static final Log log = Log.getLog(BigQueryTrace.class);
+    private Map<String, String> resultDetails;
+    private final ByteNumberFormat byteFormater = new ByteNumberFormat();
+
+    public BigQueryTrace(@NotNull BigQueryResultSet resultSet) {
+        resultDetails = new LinkedHashMap<>();
+        try {
+            Object mResultSet = BeanUtils.getInheritedFieldValue(resultSet.getOriginal(), "m_resultSet"); //$NON-NLS-1$
+            Object mDataHandler = BeanUtils.getInheritedFieldValue(mResultSet, "m_dataHandler"); //$NON-NLS-1$
+            Object mBqResults = BeanUtils.getInheritedFieldValue(mDataHandler, "m_bqResults"); //$NON-NLS-1$
+            Object mQueryResponse = BeanUtils.getInheritedFieldValue(mBqResults, "m_queryResponse"); //$NON-NLS-1$
```

Comment:
This feels really unsafe. Are there no public getters available so we can at least assume these wouldn't change in the next minor release?